controllers/machinedeletionremediation_controller.go

```diff
@@ -127,13 +172,15 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 	log.Info("request node-associated machine deletion", "machine", machine.GetName(), "node", nodeName)
 	err = r.Delete(ctx, machine)
 	if err != nil {
-		log.Error(err, "failed to delete machine associated to node", "machine", machine.GetName, "node", nodeName)
+		log.Error(err, "failed to delete machine associated to node", "machine", machine.GetName(), "node", nodeName)
+		if _, updErr := r.updateConditions(remediationFailed, remediation); updErr != nil {
```

Comment:
Did we really fail the remediation already when machine deletion fails for the first time? Deletion and remediation can succeed on next reconcile, not? Maybe we need an error count, and only mark remediation as failed (and stop requeueing) after several retries?

Reply:
I tested this in the other PR, it's hard to define how many retries is enough, however I made a wrong assumption here, that `remediationFailed` only affected `SucceedConditionType`, not `ProcessingConditionType` (the idea was Succeded:false, Processing:true), which is not :facepalm: 

I'll give retries another chance